pkg/state/types.go

```diff
@@ -579,6 +579,10 @@ func (p1 *Plugin) EqualWithOpts(p2 *Plugin, ignoreID,
 	sort.Slice(p1Copy.Tags, func(i, j int) bool { return *(p1Copy.Tags[i]) < *(p1Copy.Tags[j]) })
 	sort.Slice(p2Copy.Tags, func(i, j int) bool { return *(p2Copy.Tags[i]) < *(p2Copy.Tags[j]) })
 
+	// sort the protocols
+	sort.Slice(p1Copy.Protocols, func(i, j int) bool { return *(p1Copy.Protocols[i]) < *(p1Copy.Protocols[j]) })
```

Comment:
Protocols are never nil, correct?

Reply:
I'll verify and confirm

Reply:
sort.Slice is safe to use on empty / nil slices (verified by running in go playground)